app/controllers/fhir/r4/api_controller.rb

```diff
@@ -210,6 +210,14 @@ def create
     when 'patient'
       return if doorkeeper_authorize!(*PATIENT_WRITE_SCOPES)
 
+      # Check for duplicates
+      if request.headers['If-None-Exist'].present?
+        search_params = request.headers['If-None-Exist'].split('&').map { |p| p.split('=', 2) }.select { |p| p.size == 2 }.to_h
```

Comment:
I think `Rack::Utils.parse_nested_query` should be able to accomplish this same parse for us, and may be a little safer at dealing with any unusual cases.

Reply:
https://github.com/SaraAlert/SaraAlert/pull/1098/commits/b335e9bae20603fd40a4b1f465c206aff9f4b9ef#diff-e79e38176ee697a8541cffba23c60d9decaa69382ca5c6e875da033508e5b428R215